modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -256,9 +256,146 @@ void GOCVBGR::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
     out.post(std::move(out_arg));
 }
 
+struct GOCVY: public cv::detail::KernelTag
+{
+    using API = cv::gapi::streaming::GY;
+    static cv::gapi::GBackend backend() { return cv::gapi::streaming::backend(); }
+
+    class Actor final: public cv::gapi::streaming::IActor {
+        public:
+            explicit Actor(const cv::GCompileArgs&) {}
+            virtual void run(cv::gimpl::GIslandExecutable::IInput &in,
+                             cv::gimpl::GIslandExecutable::IOutput&out) override;
+    };
+
+    static cv::gapi::streaming::IActor::Ptr create(const cv::GCompileArgs& args)
+    {
+        return cv::gapi::streaming::IActor::Ptr(new Actor(args));
+    }
+    static cv::gapi::streaming::GStreamingKernel kernel() { return {&create}; };
+};
+
+void GOCVY::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
+                       cv::gimpl::GIslandExecutable::IOutput &out)
+{
+    const auto in_msg = in.get();
+    if (cv::util::holds_alternative<cv::gimpl::EndOfStream>(in_msg))
+    {
+        out.post(cv::gimpl::EndOfStream{});
+        return;
+    }
+
+    const cv::GRunArgs &in_args = cv::util::get<cv::GRunArgs>(in_msg);
+    GAPI_Assert(in_args.size() == 1u);
+
+    cv::GRunArgP out_arg = out.get(0);
+    auto frame = cv::util::get<cv::MediaFrame>(in_args[0]);
+    const auto& desc = frame.desc();
+
+    auto& rmat = *cv::util::get<cv::RMat*>(out_arg);
+    switch (desc.fmt)
+    {
+        case cv::MediaFormat::BGR:
+            {
+                auto view = frame.access(cv::MediaFrame::Access::R);
+                cv::Mat tmp_bgr(desc.size, CV_8UC3, view.ptr[0], view.stride[0]);
+                cv::Mat yuv;
+                cvtColor(tmp_bgr, yuv, cv::COLOR_BGR2YUV_I420);
+                rmat = cv::make_rmat<cv::gimpl::RMatAdapter>(yuv.rowRange(0, desc.size.height));
+                break;
+            }
+        case cv::MediaFormat::NV12:
+            {
+                auto view = frame.access(cv::MediaFrame::Access::R);
+                cv::Mat y_plane(desc.size, CV_8UC1, view.ptr[0], view.stride[0]);
+                rmat = cv::make_rmat<cv::gimpl::RMatMediaFrameAdapter>(y_plane, frame);
```

Comment:
Is it necessary to pass `cv::Mat` into adapter ?

Reply:
It is common adapter and is created to hold `cv::Mat` data from `cv::MediaFrame`.
But how to retrieve this `cv::Mat` from `cv::MediaFrame` is not adapter's responsibility..